### PR TITLE
5.0 PHPUnit 10 compatibility

### PR DESCRIPTION
**Pull request by Naktibalda**
New event system implemented in PHPUnit 10 forced my hand,
I replaced PHPUnit TestRunner and reporters, with Codeception TestRunner and reporters based on Codeception events.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest]
-        php: ['7.4', '8.0', '8.1']
+        php: ['8.0', '8.1']
```
Comment (TavoNiievez) [inline .github/workflows/build.yml:24]: 🥲 
First line changed that I see and it already involves a lot of work.
@Naktibalda If we are going to drop support for PHP 7.4 in the release of Codeception 5 I have to submit another couple of PRs updating the codebase to PHP 8.0.
Do you want me to do it before this is merged (which might force you to rebase) or do you want me to keep checking and do this including the changes from this PR?

File: src/Codeception/TestRunner.php [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+class TestRunner
+{
+    protected array $config = [];
+
+    public function __construct()
+    {
+        $this->config = Configuration::config();
+    }
+
+    public function prepareSuite(\PHPUnit\Framework\Test $suite, array $arguments)
```
Comment (Naktibalda) [inline src/Codeception/TestRunner.php:23]: This test contains some remnants of src/PHPUnit/Runner.php, but it isn't a Test Runner.

All this code could be moved to SuiteManager which is a sole consumer.

The real TestRunner could be extracted from `run` and `runPhpUnitTest` methods in Codeception\Suite.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Naktibalda) [reply .github/workflows/build.yml:24]: I really don't want to rebase this branch after enormous change in 5.0, too many files were moved.
Could you wait a few days?

File: src/Codeception/TestRunner.php (hunk 2 above)
Comment (Naktibalda) [reply src/Codeception/TestRunner.php]: I eliminated this class.

File: .github/workflows/build.yml (hunk 1 above)
Comment (TavoNiievez) [reply .github/workflows/build.yml:24]: Okay no problem.

File: CHANGELOG-5.x.md [hunk 3]
```diff
@@ -1,15 +1,18 @@
 #### 5.0.0
 
-* Compatible with PHPUnit 9 only (probably will change to 10-only before release)
-* Compatible with Symfony 4.4 - 6.0
-* Requires PHP 7.4 or higher
-* Merged codeception/phpunit-wrapper 9.0 branch to codeception/codeception code base.
+* Decoupled test execution and reporting from PHPUnit
+* Custom reporters implementing TestListener are no longer supported and must be converted to Exceptions.
```
Comment (DavertMik) [inline CHANGELOG-5.x.md:4]:
```suggestion
* Custom reporters implementing TestListener are no longer supported and must be converted to Extensions.
```

File: src/Codeception/Suite.php [hunk 4]
```diff
@@ -4,21 +4,288 @@
...
+            $eventType = Events::TEST_WARNING;
+        } elseif (isset($unintentionallyCoveredCodeError)) {
+            $result->addFailure(
+                $test,
+                $unintentionallyCoveredCodeError,
+                $time
+            );
+            $eventType = Events::TEST_ERROR;
+        }
+//        elseif ($this->configuration->reportUselessTests() &&
```
Comment (DavertMik) [inline src/Codeception/Suite.php:251]: because we don't have risky or useless tests, right?
maybe this code should be removed

**Review by DavertMik — APPROVED** [review]

**Comment by DavertMik** [comment]
Thanks, this is really huge refactoring!
Thank you so much!

**Review by TavoNiievez — APPROVED** [review]
Very good work

**Review by Arhell — APPROVED** [review]

File: src/Codeception/Suite.php (hunk 4 above)
Comment (Naktibalda) [reply src/Codeception/Suite.php:251]: I promised to @TavoNiievez to implement risky tests in Codeception codebase.

File: .github/workflows/build.yml (hunk 1 above)
Comment (Naktibalda) [reply .github/workflows/build.yml:24]: @TavoNiievez You are welcome to update 5.0 branch now.

**Pull request merged**